zio-http-example/src/main/scala/example/EndpointExamples.scala

```diff
@@ -36,7 +37,9 @@ object EndpointExamples extends ZIOAppDefault {
       }
     }
 
-  val routes = Routes(getUserRoute, getUserPostsRoute)
+  val openAPI = OpenAPIGen.fromEndpoints(title = "Endpoint Example", version = "1.0", getUser, getUserPosts)
+
+  val routes = Routes(getUserRoute, getUserPostsRoute) @@ Middleware.swaggerUI("docs" / "openapi", openAPI)
```

Comment:
I would like to ask the question why is this functionality a middleware? It is not modifying the behavior of existing routes just generates one or more new ones. Would not it be more logical if it is just a function returning `Routes` which can be appended to an existing `Routes`?